internal/userutils/userutils_test.go

```diff
@@ -0,0 +1,126 @@
+package userutils_test
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/userutils"
+	"github.com/ubuntu/authd/log"
+)
+
+var useSudo bool
+var gpasswdHelperPath string
+var groupFile string
+
+func TestLockAndUnlockGroupFile(t *testing.T) {
+	// This test requires either root privileges, unprivileged user namespaces, or being
+	// able to run sudo without user interaction.
+	if os.Geteuid() == 0 {
+		log.Info(context.Background(), "Running as EUID 0")
+	} else if canUseUnprivilegedUserNamespaces() {
+		log.Info(context.Background(), "Can use unprivileged user namespaces")
+	} else if canUseSudoNonInteractively() {
+		log.Info(context.Background(), "Can use sudo non-interactively")
+		useSudo = true
+	} else {
+		t.Skip("Skipping test: requires root privileges or unprivileged user namespaces")
+	}
+
+	// Build the helper binary
+	tempDir := t.TempDir()
+	gpasswdHelperPath = tempDir + "/gpasswd"
+	//nolint:gosec // G204 It's fine to pass a variable to exec.Command here
+	cmd := exec.Command("go", "build",
+		"-o", gpasswdHelperPath,
+		"-tags", "userutils_testhelpers",
+		"./testhelpers/gpasswd.go")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
```

Comment:
Why not using `CombinedOutput()` and then writing the output to the test log and in other places?

So that we respect `-test.v` option and it's properly shown in CI logs. Or in failure only.

If you want see it in real time you could use `testutils.IsVerbose()`, but I don't feel it's needed.